api/python/provisioner/commands/validator/validator.py

```diff
@@ -661,6 +664,59 @@ class AuthenticityValidator(PathValidator):
         default=None
     )
 
+    @staticmethod
+    def _convert_key_to_open_pgp_format(pub_key_path: Path) -> Path:
+        """
+        Check if GPG Public key in ASCII Armor format. If so format it to
+        OpenPGP format.
+
+        Parameters
+        ----------
+        pub_key_path: Path
+            Path to GPG public key
+
+        Returns
+        -------
+        Path:
+            path to the file with GPG public key in OpenPGP format
+
+        """
+        # NOTE: for the ASCII Armor format, please, refer to RFC4880
+        #  https://datatracker.ietf.org/doc/html/rfc4880#section-6.2
+
+        # NOTE: return given path itself if it is in OpenPGP format already
+        res = pub_key_path
+        with open(pub_key_path, "rb") as fh:
+            # NOTE: read file as binary file since OpenPGP is binary format
+            content = fh.readlines()
+            armor_header = content[0]
+            armor_tail = content[-1]
+
+        # NOTE: we check that the armor header and armor tail in binary
+        #  representation exist in the first and the last line of
+        #  the pub key file.
+        if (ARMOR_HEADER.encode() in armor_header
+                and ARMOR_TAIL.encode() in armor_tail):
+            # NOTE: it means that provided public key is in ASCII Armor format
+            cmd = f"gpg --yes --dearmor {pub_key_path.resolve()}"
+            try:
+                # NOTE: by default gpg tool converts the given file to the file
+                #  with the same name + '.gpg' extension at the end.
+                #  Directory is the same
+                cmd_run(cmd, targets=local_minion_id(),
+                        fun_kwargs=dict(python_shell=True))
+            except Exception as e:
+                logger.error("Can't convert ASCII Armor GPG public key "
```

Comment:
could you mention path to pub key here?

Reply:
Yes, sure. I added given public key into error message output.